File: programmers/python/입문100/12.4_소인수분해★/12-4.py
def solution(n):
    factors = []
    d = 2
    while d * d <= n:   # √n 까지만 검사
        if n % d == 0:
            factors.append(d)
            while n % d == 0:  # 중복 제거 위해 같은 소인수는 전부 나눠줌
                n //= d
        d += 1
    if n > 1:   # 마지막에 남은 소수가 있다면 추가
        factors.append(n)
    return factors

File: programmers/java/기초124/12.4_2의영역/12_4.java
class Solution {
    public int[] solution(int[] arr) {
        // int[] answer = {};
        // return answer;
        //////////////////////////////////////
        int first = -1;
        int last = -1;

        for (int i = 0; i < arr.length; i++){
            if (arr[i] == 2) {
                if (first == -1) first = i;
                last = i;
            }
        }

        if (first == -1) {
            return new int[]{-1};
        }

        int[] answer = new int[last - first + 1];
        System.arraycopy(arr, first, answer, 0, answer.length);
        //////////////////////////////////////
        return answer;
    }
}